src/krux/wallet.py

```diff
@@ -112,9 +125,13 @@ def obtain_addresses(self, i=0, limit=None, branch_index=0):
         """Returns an iterator deriving addresses (default branch_index is receive)
         for the wallet up to the provided limit"""
         starting_index = i
+        if self.key:
+            network = self.key.network
+        else:
+            network = NETWORKS[Settings().wallet.network]
```

Comment:
Here again, why not use the descriptor derivation path info? It would serve for the two cases, when loading the wallet and when loading only the descriptor

Reply:
While there are preferred derivation path standards which differ based on the network, I feel the path is a much softer hint of network compared to the version bytes on the front of the xpub.  It really feels like embit.descriptor deserves to have a "network" attribute, else possibly krux.wallet needs one... but for now I just want to get it working for single-sig and multisig address listings.

As a trial, I'll use the first network I find in the order of:
1) whatever the krux user loaded their mnemonic with (which comes from settings but they may have changed it),
2) the first xpub in the descriptor, via the version bytes and a value-to-key lookup of embit.networks.NETWORKS
3) if for some reason it's not found, but I cannot imagine why, the network from settings.

...but if krux.wallet had a network attribute that got filled in by parsing the descriptor or by the user when they load the mnemonic, it would be much cleaner and faster for scanning many pages of addresses based on descriptor alone... and could further serve to fail/warn when descriptor network and krux user's choice don't match... with the downside (I'll argue upside) of not being able to load a mainnet xpub while setup for testnet and seeing testnet addresses instead of an error "descriptor is not for this network".

Reply:
In my opinion, this feature should focus on the descriptor only. That's why I would call it "Descriptor Addresses".
Krux will use data within the descriptor to export addresses.
Incomplete descriptors, without enough data to derive addresses, should not be allowed.
Assembling wallets from pieces of information from multiple sources doesn't seem like a good idea to me.

Reply:
Initially, I was only considering the new feature in isolation. I hadn't realized that you were also addressing the scenario of descriptor loading with an already loaded key, and how these two situations are naturally connected in code. Your proposed approach seems ideal to me.